src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -106,6 +111,14 @@ public ILImporter(WebAssemblyCodegenCompilation compilation, MethodDesc method,
             _llvmFunction = GetOrCreateLLVMFunction(mangledName, method.Signature);
             _builder = LLVM.CreateBuilder();
             _pointerSize = compilation.NodeFactory.Target.PointerSize;
+
+            // Get debug metadata for the uninstantiated version of the method
+            MethodDesc typicalMethod = _method.GetTypicalMethodDefinition();
```

Comment:
Call `GetDebugInfo` on `Compilation` to get the debug information. It will opt you into conveniences such as respecting the `--ildump` and `-g` command line options, or dealing with methods whose bodies were replaced.

Reply:
Done.